### PR TITLE
{Network} Network support extendedLocation

### PR DESCRIPTION
**Pull request by msyyc**
**Description**

Similar link:
storage : https://github.com/Azure/azure-cli/pull/17528
compute: https://github.com/Azure/azure-cli/pull/17522
'az network vnet-gateway create' need new swagger : https://github.com/Azure/azure-rest-api-specs/pull/13864/files
(RE: virtualNetworkGateway missing extendedLocation in swagger)

**Testing Guide**

NetworkExtendedLocation
**History Notes**

[Network]`az network lb create`: Add new parameter `--edge-zone`
[Network]`az network nic create`: Add new parameter `--edge-zone`
[Network]`az network private-endpoint create`: Add new parameter `--edge-zone`
[Network]`az network private-link-service create`: Add new parameter `--edge-zone`
[Network]`az network public-ip create`: Add new parameter `--edge-zone`
[Network]`az network public-ip prefix create`: Add new parameter `--edge-zone`
[Network]`az network vnet create`: Add new parameter `--edge-zone`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 1]
```diff
@@ -111,6 +111,7 @@ def load_arguments(self, _):
         help='Space-separated list of availability zones into which to provision the resource.',
         choices=['1', '2', '3']
     )
+    edge_zone = CLIArgumentType(help='The name of edge zone.', is_preview=True, min_api='2020-08-01')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:114]: You could use this type for edge zone and add min_api version for your parameter
https://github.com/Azure/azure-cli/blob/3f3aafdd3fec06d66d318eeac7cbc0e70f124107/src/azure-cli-core/azure/cli/core/commands/parameters.py#L251

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 2]
```diff
@@ -1925,6 +1933,7 @@ def load_arguments(self, _):
         vnet_help = "Name or ID of an existing virtual network which has a subnet named 'GatewaySubnet'."
         c.argument('virtual_network', options_list='--vnet', help=vnet_help)
         c.argument('vpn_gateway_generation', arg_type=get_enum_type(['Generation1', 'Generation2']), min_api='2019-07-01', help='The generation for the virtual network gateway. vpn_gateway_generation should not be provided if gateway_type is not Vpn.')
+        c.ignore('edge_zone')  # wait for service ready
```
[the remark this answers is not in the document]
Comment (msyyc) [reply src/azure-cli/azure/cli/command_modules/network/_params.py:1936]: 'az network vnet-gateway create' need new swagger : https://github.com/Azure/azure-rest-api-specs/pull/13864/files
(RE: virtualNetworkGateway missing extendedLocation in swagger)

[the remark this answers is not in the document]
Comment (msyyc) [reply src/azure-cli/azure/cli/command_modules/network/_params.py:1936]: It is ignored. So customer could not use it.

[the remark this answers is not in the document]
Comment (kairu-ms) [reply src/azure-cli/azure/cli/command_modules/network/_params.py:1936]: > It is ignored. So customers could not use it.

I think customer can see it in command help message. Why not expose it when service is ready?

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**